src/internals/StreamHttpReply.cpp

```diff
@@ -132,3 +135,29 @@ String ArduinoHttpServer::StreamHttpErrorReply::getJsonBody(const String& data)
 
    return body;
 }
+
+//------------------------------------------------------------------------------
+//                             Class Definition
+//------------------------------------------------------------------------------
+
+ArduinoHttpServer::StreamHttpAuthenticateReply::StreamHttpAuthenticateReply(Stream& stream, const String& contentType) :
+   AbstractStreamHttpReply(stream, contentType, "401")
+{
+
+}
+
+void ArduinoHttpServer::StreamHttpAuthenticateReply::send()
+{
+   // Read away remaining bytes.
+   while(getStream().read()>=0);
+
+   DEBUG_ARDUINO_HTTP_SERVER_PRINT("Printing Reply ... ");
+   getStream().println(AHS_F("HTTP/1.1 401 Unauthorized"));
+   getStream().println(AHS_F("WWW-Authenticate: Basic realm=\"Login Required\""));
+   getStream().println(AHS_F("Connection: close"));
+   getStream().println(AHS_F(""));
+   getStream().println(AHS_F("<HTML><HEAD><TITLE>401 Unauthorized</TITLE></HEAD><BODY BGCOLOR=\"#cc9999\"><H4>401 Unauthorized</H4>Authorization required.</BODY></HTML>"));
```

Comment:
Would you be willing to make the HTML lowercase? I.e. identical to HttpErrorReply's HTML markup casing? I would also like to keep the allocated amount of characters to a minimum, meaning the bgcolor is maybe unnecessary.

Reply:
Will fix.

Reply:
<html><head><title..